File: src/SDScorer.java
import java.util.ArrayList;
import java.util.OptionalDouble;

public class SDScorer {
    public ArrayList<Double> sdScorer(ArrayList<Double> List, int sample) {
        OptionalDouble weirdAverage = List.stream().mapToDouble(a -> a).average();
        double average = weirdAverage.getAsDouble();
        double difference = 0;
        for (Double item : List) {
            double value = item - average;
            difference += Math.pow(value, 2);
        }
        double sd = Math.sqrt(difference / (List.size() - sample));
        ArrayList<Double> sdScored = new ArrayList<>();
        for (Double item : List) {
            double change = item - average;
            double sdScore = change / sd;
            sdScored.add(sdScore);
        }
        return sdScored;
    }
}

File: src/SDScorer.kt
import kotlin.math.pow
import kotlin.math.sqrt

class SDScorer {
    fun sdScorer(list: List<Double>, sample: Int): List<Double> {
        val average = list.average()
        var difference = 0.0
        for (item in list) {
            val value = item - average
            difference += value.pow(2)
        }
        val sd = sqrt(difference / (list.size - sample))
        val sdScored = mutableListOf<Double>()
        for (item in list) {
            val change = item - average
            val sdScore = change / sd
            sdScored.add(sdScore)
        }
        return sdScored
    }
}